### ISSUE TITLE
"No module named tmpl" exception- when running under 2.7 ubuntu 12.04 with supervisord

### ISSUE THREAD

**Issue by gsvitak** [onelogin/python-saml#163]
Hello

As I have stated in the past, thanks for the great work. 

I am using the latest python-saml in production. We have a gunicorn server running under supervisord. When a worker times out or crashes,supervisor attempts to automatically restart the gunicorn worker. The worker restart is the desired behavior.

However when the worker attempt to restart, I get the following stack trace and all of the worker crash.

```
  File "/home/rbadmin/checkouts/trailhead/python-code/trailhead/model/authentication/logout.py", line 6, in <module>
    from onelogin.saml2.auth import OneLogin_Saml2_Auth
  File "/home/rbadmin/.virtualenvs/trailhead/local/lib/python2.7/site-packages/onelogin/saml2/auth.py", line 20, in <module>
    from onelogin.saml2.settings import OneLogin_Saml2_Settings
  File "/home/rbadmin/.virtualenvs/trailhead/local/lib/python2.7/site-packages/onelogin/saml2/settings.py", line 20, in <module>
    from onelogin.saml2.metadata import OneLogin_Saml2_Metadata
  File "/home/rbadmin/.virtualenvs/trailhead/local/lib/python2.7/site-packages/onelogin/saml2/metadata.py", line 17, in <module>
    from onelogin.saml2.utils import OneLogin_Saml2_Utils
  File "/home/rbadmin/.virtualenvs/trailhead/local/lib/python2.7/site-packages/onelogin/saml2/utils.py", line 32, in <module>
    from dm.xmlsec.binding.tmpl import EncData, Signature
```

I can also reproduce the error by having the gunicorn server running and then open a python console. If I execute the following, the error occurs as well.

```
Python 2.7.3 (default, Feb 27 2014, 20:00:17)
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import dm.xmlsec.binding._xmlsec
>>> from dm.xmlsec.binding.tmpl import EncData, Signature
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named tmpl
>>>
```

If I shutdown the gunicorn process via supervisord, I have to reinstall python-saml via 

`pip install pyth-saml --ignore-installed`

to force the library to work properly. After the reinstall is completed, I can restart the gunicorn workers via supervisor

In my opinion, I think that the python binding is locked in supervisor process and not allowed to be imported until the process releases the lock.

I have seen similar behavior on Ubuntu 15.04 as well. 

Basically I am curious how can I troubleshoot the issue? Do you have any suggestions on to turn on some logging on dm.xmlsec to figure how to fix the issue?

I have asked about this issue in the past but now have more information and wanted to see if you have a suggestion.

Thanks
G

**Comment by pitbulk** [comment]
No much time ago we changed the way xmlsec was initialized [PR 149](https://github.com/onelogin/python-saml/pull/149), but no idea why the tmpl module turns no accessible. It seems an dm.xmlsec.binding issue and I don't know how to fix it. Maybe you can try to contact its author, Dieter Maurer, and report it.

I wonder if you tried to use [python3-saml](https://github.com/onelogin/python3-saml/) that is a py3 compatible version of python-saml that uses a different xmlsec wrapper: [python-xmlsec](https://github.com/mehcode/python-xmlsec).

**Comment by gsvitak** [comment]
thank you. Sorry I did not understand that the other project was compatible with 2.7. I read the description quickly and thought it was only for python 3. Does python3 support the HTTP-POST binding for SSO?

Thanks
Greg

**Comment by pitbulk** [comment]
HTTP-POST binding for SSO only for the ACS, rest is HTTP-Redirect binding.
(python3-saml is like python-saml but compatible with py3 and use a different xmlsec wrapper, but the functionality is the same).

**Comment by gsvitak** [comment]
Thank you. We will integrate python3-saml. FYI I already forked python3-saml and added in HTTP-POST binding for SSO and DigestMethodAlgorithm on the security settings. We are also adding in support for a configurable SPNameID for SAMLLogout to support ADFS 3.0 with the latest Netscaler as Web Proxy. Let me know if you would like pull requests.

**Comment by pitbulk** [comment]
A PR is always welcome, I don't know if we will support that soon on python3-saml but meanwhile maybe others will be able to integrate in their environments.
